### ISSUE TITLE
Default button lacks a 'pressed' state

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#3620]
## Environment
React Storybook

> Operating system
Windows 7

> Browser
Chrome and Firefox

## Detailed description
There appears to be no "pressed" button state.  It may be just a shortcoming of the demo site, because none of the buttons are actually executable; however, there should be  a differentiation between the focus indicator and the state of a button after it has been activated.

> What did you expect to happen?

After I select a button, it shows up as pressed in some way, so that I can distinguish which of the buttons is currently activated.

> What happened instead?

The buttons do not visually alter their appearance when pressed.

> What WCAG 2.1 checkpoint does the issue violate?

[Non-text Contrast](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast)
https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast
>Visual information required to identify user interface components and states,

## Steps to reproduce the issue

1. Expand the buttons area of the left navigation and select Default
2. Choose Open canvas in new tab
3. Put focus on a button
4. Press Enter

## Additional information
![image](https://user-images.githubusercontent.com/14298245/62267618-93c9cc00-b3e1-11e9-9d74-a55eb61146d9.png)
Button labelled 'Button' has focus. Looks exactly the same whether or not it has been activated (pressed)

**Comment by asudoh** [comment]
Hi 👋 have you tried space key instead of enter key? Native HTML buttons are activated by space key instead of enter key.

**Comment by tw15egan** [comment]
I think buttons will only have an active state on `Enter` if they have a `type="submit"` and a part of a `Form`.  Here is an example of the active state on mouse click/space key: 

![button](https://user-images.githubusercontent.com/11928039/62322251-7df7ed80-b459-11e9-8717-8e31e264d1cd.gif)

Code where the active state is set: 
https://github.com/carbon-design-system/carbon/blob/master/packages/components/src/components/button/_mixins.scss#L87

Going to close this as there is a defined active state
